projects/com.oracle.truffle.llvm.nodes/src/com/oracle/truffle/llvm/nodes/asm/syscall/LLVMAMD64Memory.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright (c) 2017, Oracle and/or its affiliates.
+ *
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without modification, are
+ * permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice, this list of
+ * conditions and the following disclaimer.
+ *
+ * 2. Redistributions in binary form must reproduce the above copyright notice, this list of
+ * conditions and the following disclaimer in the documentation and/or other materials provided
+ * with the distribution.
+ *
+ * 3. Neither the name of the copyright holder nor the names of its contributors may be used to
+ * endorse or promote products derived from this software without specific prior written
+ * permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS
+ * OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF
+ * MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ * COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+ * EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE
+ * GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED
+ * AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+ * NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED
+ * OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package com.oracle.truffle.llvm.nodes.asm.syscall;
+
+import com.oracle.truffle.llvm.runtime.LLVMAddress;
+import com.oracle.truffle.llvm.runtime.memory.LLVMMemory;
+
+public class LLVMAMD64Memory {
+    public static final int PROT_READ = 0x1;
+    public static final int PROT_WRITE = 0x2;
+    public static final int PROT_EXEC = 0x4;
+    public static final int PROT_SEM = 0x8;
+    public static final int PROT_NONE = 0x0;
+    public static final int PROT_GROWSDOWN = 0x01000000;
+    public static final int PROT_GROWSUP = 0x02000000;
+
+    public static final int MAP_SHARED = 0x01;
+    public static final int MAP_PRIVATE = 0x02;
+    public static final int MAP_TYPE = 0x0f;
+    public static final int MAP_FIXED = 0x10;
+    public static final int MAP_ANONYMOUS = 0x20;
+    public static final int MAP_UNINITIALIZED = 0x4000000;
+
+    public static long brk(@SuppressWarnings("unused") LLVMAddress ptr) {
+        return -LLVMAMD64Error.ENOSYS;
```

Comment:
Could you document the current assumptions if a system does not implement the expected semantics (in this and other cases)?

Reply:
Correct code has to check for errors after every syscall. If a syscall is not implemented, it has to return ENOSYS. Therefore, this is the correct behavior for unimplemented syscalls. Real world applications ignore that sometimes, and things will break.

Reply:
Could you add some comment like "// not supported"?